### PR TITLE
feat: add data urls and possibility to send data via sumologicexporter

### PR DESCRIPTION
**Pull request by pmalek-sumo**

### CODE REVIEW

File: pkg/exporter/sumologicexporter/exporter.go [hunk 1]
```diff
@@ -16,15 +16,16 @@ package sumologicexporter
...
 	"net/http"
-	"reflect"
+	"strings"
 
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/consumer/consumererror"
 	"go.opentelemetry.io/collector/consumer/pdata"
 	"go.opentelemetry.io/collector/exporter/exporterhelper"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/sumologicextension"
```
Comment (pmatyjasek-sumo) [inline pkg/exporter/sumologicexporter/exporter.go:28]: shouldn't we use local extension?

File: pkg/exporter/sumologicexporter/exporter.go [hunk 2]
```diff
@@ -342,20 +349,37 @@ func (se *sumologicexporter) pushMetricsData(ctx context.Context, md pdata.Metri
...
+	)
+	for _, e := range host.GetExtensions() {
+		v, ok := e.(*sumologicextension.SumologicExtension)
+		if ok {
+			ext = v
+			foundSumoExt = true
+			break
 		}
+	}
+	// if !found {
```
Comment (pmatyjasek-sumo) [inline pkg/exporter/sumologicexporter/exporter.go:364]: Why do we need this commented out code?

File: pkg/exporter/sumologicexporter/exporter.go [hunk 3]
```diff
@@ -342,20 +349,37 @@ func (se *sumologicexporter) pushMetricsData(ctx context.Context, md pdata.Metri
...
-		out := m.Func.Call([]reflect.Value{reflect.ValueOf(e)})
-		fmt.Printf("Collector ID: %s\n", out[0].String())
+	// If we're using sumologicextension as authentication extension and
+	// endpoint was not set then send data on a collector generic ingest URL
+	// with authentication set by sumologicextension.
+	if se.config.HTTPClientSettings.Endpoint == "" &&
+		// TODO: is there a better way than strings.Prefix of auth name?
+		strings.HasPrefix(se.config.HTTPClientSettings.Auth.AuthenticatorName, "sumologic") &&
+		foundSumoExt {
+		se.dataUrlLogs = ext.BaseUrl() + "/api/v1/collector/logs"
```
Comment (pmatyjasek-sumo) [inline pkg/exporter/sumologicexporter/exporter.go:375]: make `/api/v1/collector/logs` constant

File: pkg/exporter/sumologicexporter/exporter.go [hunk 4]
```diff
@@ -342,20 +349,37 @@ func (se *sumologicexporter) pushMetricsData(ctx context.Context, md pdata.Metri
...
+		se.dataUrlMetrics = ext.BaseUrl() + "/api/v1/collector/metrics"
```
Comment (pmatyjasek-sumo) [inline pkg/exporter/sumologicexporter/exporter.go:376]: make `/api/v1/collector/metrics` constant

File: pkg/exporter/sumologicexporter/sender.go [hunk 5]
```diff
@@ -108,63 +114,47 @@ func (s *sender) send(ctx context.Context, pipeline PipelineType, body io.Reader
...
-		req.Header.Add(headerFields, flds.string())
-	case MetricsPipeline:
-		switch s.config.MetricFormat {
-		case PrometheusFormat:
-			req.Header.Add(headerContentType, contentTypePrometheus)
-		case Carbon2Format:
-			req.Header.Add(headerContentType, contentTypeCarbon2)
-		case GraphiteFormat:
-			req.Header.Add(headerContentType, contentTypeGraphite)
+func (s *sender) getRequest(ctx context.Context, pipeline PipelineType, data io.Reader) (*http.Request, error) {
```
Comment (pmatyjasek-sumo) [inline pkg/exporter/sumologicexporter/sender.go:137]: I think `createRequest` or `buildRequest` will be more intuitive name

File: pkg/extension/sumologicextension/config.go [hunk 6]
```diff
@@ -22,10 +22,16 @@ import (
 type Config struct {
 	config.ExtensionSettings `mapstructure:"-"`
 
+	Credentials credentials `mapstructure:",squash"`
```
Comment (pmatyjasek-sumo) [inline pkg/extension/sumologicextension/config.go:25]: Missing description

File: pkg/extension/sumologicextension/extension.go [hunk 7]
```diff
@@ -29,153 +29,46 @@ import (
...
 		return nil, errors.New("collector name is unset")
 	}
 
-	return &sumologicExtension{
+	return &SumologicExtension{
 		// TODO: don't hardcode
 		baseUrl: niteBaseUrl,
-		conf:    conf,
-		logger:  logger,
+		// baseUrl: "http://localhost:3000",
```
Comment (pmatyjasek-sumo) [inline pkg/extension/sumologicextension/extension.go:52]: commented out code

**Review by pmatyjasek-sumo — CHANGES_REQUESTED** [review]

File: pkg/exporter/sumologicexporter/exporter.go (hunk 1 above)
Comment (pmalek-sumo) [reply pkg/exporter/sumologicexporter/exporter.go:28]: We do, take a look at the go.mod

https://github.com/SumoLogic/sumologic-otel-collector/pull/3/files#diff-810660f89ab0e3e51263333e86eba41eefc5fdbbedc135bcdb2dc3bf115e6455R13

**Review by pmatyjasek-sumo — APPROVED** [review]

**Pull request merged**